Python/ceval.c

```diff
@@ -2484,6 +2484,19 @@ _PyEval_EvalFrameDefault(PyFrameObject *f, int throwflag)
             DISPATCH();
         }
 
+        case TARGET(BUILD_LIST_PREALLOC): {
+            PyObject *list, *target = GETLOCAL(oparg);
+            Py_INCREF(target);
+            Py_ssize_t size = PyObject_LengthHint(target, 2);
+            Py_DECREF(target);
+            if (size < 0)
+                goto error;
```

Comment:
IMHO this should raise a `ValueError` or something other than OverflowError with an error message for an average Python user.

Reply:
Have raised that separately in #12720 as it's out-of-scope for this change and impact other things